src/include/migraphx/tracer.hpp

```diff
@@ -1,33 +1,43 @@
 #ifndef MIGRAPHX_GUARD_RTGLIB_TRACER_HPP
 #define MIGRAPHX_GUARD_RTGLIB_TRACER_HPP
 
-#include <ostream>
+#include <fstream>
 #include <migraphx/functional.hpp>
 #include <migraphx/config.hpp>
-
+#include <migraphx/filesystem.hpp>
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 
 struct tracer
 {
     tracer() {}
 
-    tracer(std::ostream& s) : os(&s) {}
+    tracer(std::string dump_directory) : dump_dir(std::move(dump_directory)), counter(0) {}
```

Comment:
The tracer should be able to write to an `ostream`, otherwise we cant write to `cout` or `cerr`.